docs/runtime-code-typescript-setup.md

```diff
@@ -0,0 +1,146 @@
+
+## Prerequisites
+
+You will need to have these tools installed to work with TypeScript for your project:
+
+* Node v14 (active LTS) or greater.
+* Basic UNIX tools or knowledge on the Windows equivalents.
+
+The TypeScript compiler and other dependencies will be fetched with NPM.
+
+## Initialize the Project
+
+These steps will set up a workspace to write all your project code to be run by the game server.
+
+1. Define the folder name that will be the workspace for the project. In this case we'll use "ts-project".
+
+    ``` shell
+    mkdir -p ts-project/{src,build}
+    cd ts-project
+    ```
+
+2. Use NPM to set up the Node dependencies in the project. Install the TypeScript compiler.
+
+    ``` shell
+    npm init -y
+    npm install --save-dev typescript
+    ```
+
+3. Use the TypeScript compiler installed to the project to set up the compiler options.
+
+    ``` shell
+    npx tsc --init
+    ```
+
+    You'll have a "tsconfig.json" file describe the available options that can be added run on the TypeScript compiler. When you've trimmed the commented out entries and updated it a minimal file will look something like:
+
+    ``` json
+    {
+      "compilerOptions": {
+        "target": "es5",
+        "strict": true,
+        "esModuleInterop": true,
+        "skipLibCheck": true,
+        "forceConsistentCasingInFileNames": true
+      }
+    }
+    ```
+
+    Add this configuration option to the `"compilerOptions"` block:
+
+    ``` json
+    "outFile": "./build/index.js",
+    ```
+
+4. Add the Nakama runtime types as a dependency to the project and configure the compiler to find the types.
+
+    ``` shell
+    npm i 'https://github.com/heroiclabs/nakama-common'
+    ```
+
+    Add this configuration option to the `"compilerOptions"` block of the "tsconfig.json" file:
+
+    ``` json
+    "typeRoots": [
+      "./node_modules"
+    ],
+    ```
+
+    This completes the setup and your project should look similar to this layout:
+
+    ``` shell
+    .
+    ├── build
+    ├── node_modules
+    │   ├── nakama-runtime
+    │   └── typescript
+    ├── package-lock.json
+    ├── package.json
+    ├── src
+    └── tsconfig.json
+    ```
+
+## Develop Code
+
+We'll write some simple code and compile it to JavaScript so it can be run by the game server.
+
+All code must start execution from a function that the game server looks for in the global namespace at startup. This function must be called `"InitModule"` and is how you register RPCs, before/after hooks, and other event functions managed by the server.
+
+The code below is a simple Hello World example which uses the `"Logger"` to write a message. Name the source file "main.ts" inside the "src" folder. You can write it in your favourite editor or IDE.
+
+``` typescript
+let main: nkruntime.InitModule = function(ctx: nkruntime.Context, logger: nkruntime.Logger, nk: nkruntime.Nakama, initializer: nkruntime.Initializer) {
+    logger.info("Hello World!");
+}
+
+let InitModule = main;
```

Comment:
Why not skip the `main` declaration and just set the lambda to `InitModule`?

Reply:
I was trying to keep the code example horizontal flow shorter but you're right it makes little difference.